src/main/java/net/minecraftforge/event/AnvilUpdateEvent.java

```diff
@@ -19,52 +19,146 @@
         this.output = ItemStack.EMPTY;
         this.name = name;
+        this.player = player;
         this.setCost(cost);
         this.setMaterialCost(0);
     }
 
-    @Nonnull
-    public ItemStack getLeft() { return left; }
-    @Nonnull
-    public ItemStack getRight() { return right; }
-    public String getName() { return name; }
-    @Nonnull
-    public ItemStack getOutput() { return output; }
-    public void setOutput(@Nonnull ItemStack output) { this.output = output; }
-    public int getCost() { return cost; }
-    public void setCost(int cost) { this.cost = cost; }
-    public int getMaterialCost() { return materialCost; }
-    public void setMaterialCost(int materialCost) { this.materialCost = materialCost; }
+    /**
+     * @return ihe Item in the left input (leftmost) anvil slot.
+     */
+    public ItemStack getLeft()
+    {
+        return left;
+    }
+
+    /**
+     * @return The item in the right input (center) anvil slot.
+     */
+    public ItemStack getRight()
+    {
+        return right;
+    }
+
+    /**
+     * This is the name as sent by the client.  It may be null if none has been sent. <br>
+     * If empty, it indicates the user wishes to clear the custom name from the item.
+     * @return The name that the output item will be set to, if applicable.
+     */
+    @Nullable
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * This is the output as a result of this event, not of the two items. <br>
+     * If you are the first receiver of this event, it is guaranteed to be empty. <br>
+     * It will only be non-empty if changed by an event handler.
+     * @return The item in the output (rightmost) anvil slot.
+     */
+    public ItemStack getOutput() 
+    {
+        return output;
+    }
+
+    /**
+     * Sets the output slot to a specific itemstack.
+     * @param output The stack to change the output to.
+     */
+    public void setOutput(ItemStack output)
+    {
+        this.output = output;
+    }
+
+    /**
+     * This is the level cost of this anvil operation. <br> 
+     * When unchanged, it is guaranteed to be left.getRepairCost() + right.getRepairCost().
+     * @return The level cost of this anvil operation.
+     */
+    public int getCost()
+    {
+        return cost;
+    }
+
+    /**
+     * Changes the level cost of this operation.
+     * @param cost The new level cost.
+     */
+    public void setCost(int cost)
+    {
+        this.cost = cost;
+    }
+
+    /**
+     * The material cost is how many right inputs are consumed.
+     * @return The material cost of this anvil operation.
+     */
+    public int getMaterialCost()
+    {
+        return materialCost;
+    }
+
+    /**
+     * Sets how many right inputs are consumed. <br>
+     * A material cost of zero consumes the entire stack. <br>
+     * A material cost higher than the count of the right stack
+     * consumes the entire stack. <br>
+     * It does not block the operation.
```

Comment:
What is "it" here? What is not being blocked?

Reply:
Yeah that's vague.  The output being obtainable is not prevented by the right stack count being lower than the material cost.

Essentially the material cost has no bearing on whether or not the output is available.